lib/litani.py

```diff
@@ -67,6 +68,48 @@ class TimeoutExpired(Exception):
     pass
 
 
+@dataclasses.dataclass
+class CacheDir:
+    cache_dir_path: pathlib.Path = None
+
+    @staticmethod
+    def get(lookup_path=os.getcwd()):
+        if CacheDir.cache_dir_path:
+            return CacheDir.cache_dir_path
+        def cache_pointer_dirs():
+            current = pathlib.Path(lookup_path).resolve(strict=True)
+            yield current
+            while current.parent != current:
+                current = current.parent
+                yield current
+
+        for possible_dir in cache_pointer_dirs():
+            logging.debug(
+                "Searching for cache pointer in %s", possible_dir)
+            possible_pointer = possible_dir / CACHE_POINTER
+            try:
+                if possible_pointer.exists():
```

Comment:
In a separate commit, could you refactor this code so that it's not so deeply nested?

```
if not possible_pointer.exists():
    continue

# de-indent the rest of the method
```